File: src/main/python/net/coderextreme/data/sphere.py
from x3dpsail import *
X3D0 = X3D()
X3D0.setProfile("Interchange")
X3D0.setVersion("3.3")
head1 = head()
component2 = component()
component2.setName("Scripting")
component2.setLevel(1)

head1.addComponent(component2)
meta3 = meta()
meta3.setName("title")
meta3.setContent("sphere.x3d")

head1.addMeta(meta3)
meta4 = meta()
meta4.setName("creator")
meta4.setContent("John Carlson")

head1.addMeta(meta4)
meta5 = meta()
meta5.setName("generator")
meta5.setContent("manual")

head1.addMeta(meta5)
meta6 = meta()
meta6.setName("identifier")
meta6.setContent("https://coderextreme.net/X3DJSONLD/sphere.x3d")

head1.addMeta(meta6)
meta7 = meta()
meta7.setName("description")
meta7.setContent("a sphere")

head1.addMeta(meta7)

X3D0.setHead(head1)
Scene8 = Scene()
Group9 = Group()
Shape10 = Shape()
Appearance11 = Appearance()
Material12 = Material()
Material12.setDiffuseColor([1,1,1])

Appearance11.setMaterial(Material12)

Shape10.setAppearance(Appearance11)
Sphere13 = Sphere()

Shape10.setGeometry(Sphere13)

Group9.addChildren(Shape10)

Scene8.addChildren(Group9)

X3D0.setScene(Scene8)
X3D0.toFileX3D("../data/sphere_RoundTrip.x3d")

File: src/main/cplusplus/net/coderextreme/data/sphere.cpp
/* delete next 5 lines for Visual Studio */
#define FALSE false
#define TRUE true
#define WINAPI
#define AFX_EXT_CLASS
#define EXPORT32
#include "C:/x3d-code/www.web3d.org/x3d/stylesheets/cpp/concretes.h"
int main(int argc, char ** argv) {
CX3DScene m_pScene;
CBrowser browser = X3D.getBrowser();
CX3D* X3D0 = new CX3D();
CGroup* group = (CGroup*)(m_pScene.createNode("Group"));
group->addChildren(X3D0);
X3D0->setProfile("Interchange");
X3D0->setVersion("4.0");
Chead* head1 = new Chead();
Cmeta* meta2 = new Cmeta();
meta2->setName("title");
meta2->setContent("sphere.x3d");
head1->addMeta(*meta2);

Cmeta* meta3 = new Cmeta();
meta3->setName("creator");
meta3->setContent("John Carlson");
head1->addMeta(*meta3);

Cmeta* meta4 = new Cmeta();
meta4->setName("generator");
meta4->setContent("manual");
head1->addMeta(*meta4);

Cmeta* meta5 = new Cmeta();
meta5->setName("identifier");
meta5->setContent("https://coderextreme.net/X3DJSONLD/src/main/data/sphere.x3d");
head1->addMeta(*meta5);

Ccomponent* component6 = new Ccomponent();
component6->setName("Scripting");
component6->setLevel(1);
head1->addComponent(*component6);

X3D0->setHead(*head1);

CScene* Scene7 = new CScene();
CGroup* Group8 = (CGroup *)(m_pScene.createNode("Group"));
CShape* Shape9 = (CShape *)(m_pScene.createNode("Shape"));
CAppearance* Appearance10 = (CAppearance *)(m_pScene.createNode("Appearance"));
CMaterial* Material11 = (CMaterial *)(m_pScene.createNode("Material"));
Material11->setDiffuseColor(new float[3]{1,1,1});
Appearance10->setMaterial(*Material11);

Shape9->setAppearance(*Appearance10);

CSphere* Sphere12 = (CSphere *)(m_pScene.createNode("Sphere"));
Shape9->setGeometry(Sphere12);

Group8->addChildren(*Shape9);

group->addChildren(*Group8);

X3D0->setScene(*Scene7);

m_pScene.addRootNode(group);
X3D0->toXMLString();
}
